connector_test.go

```diff
@@ -123,37 +117,104 @@ var volumeConfig = `
 }
 `
 
-func TestSimpleVolume(t *testing.T) {
-	logger := log.NewTestLogger(t)
-	fileContent, err := os.ReadFile("./tests/volume/test_file.txt")
-	assert.NoError(t, err)
-
-	connector, _ := getConnector(t, volumeConfig)
-	cwd, err := os.Getwd()
-	assert.NoError(t, err)
-	// disable selinux on the test folder in order to make the file readable from within the container
-	cmd := exec.Command("chcon", "-Rt", "svirt_sandbox_file_t", fmt.Sprintf("%s/tests/volume", cwd)) //nolint:gosec
-	err = cmd.Run()
-	if err != nil {
-		logger.Warningf("failed to set SELinux permissions on folder, chcon error: %s, this may cause test failure if SELinux is enabled.", err.Error())
-	}
-
-	container, err := connector.Deploy(
+// bindMountHelper is a helper function which tests plugins with a file
+// bind-mounted inside the container.  Options for the mount and the expected
+// outcome of the test are provided by parameters.  The test creates a
+// temporary file containing appropriate content, configures that file to be
+// mounted inside the container, and then starts the plugin; the test then
+// tells the plugin to output the contents of the mapped file and checks it
+// against the value originally written to the file.
+func bindMountHelper(t *testing.T, options string, expectedPass bool) {
+	fileContent := fmt.Sprintf("bind mount test with option %q\n", options)
+	mountFile := assert.NoErrorR[*os.File](t)(os.CreateTemp("", "bind_mount_test_*.txt"))
+	t.Cleanup(func() { assert.NoError(t, os.Remove(mountFile.Name())) })
+	assert.NoErrorR[int](t)(mountFile.WriteString(fileContent))
+	assert.NoError(t, mountFile.Close())
+	connector, _ := getConnector(t, fmt.Sprintf(volumeConfig, mountFile.Name(), options))
+
+	// Run the plugin
+	container := assert.NoErrorR[deployer.Plugin](t)(connector.Deploy(
 		context.Background(),
-		"quay.io/arcalot/podman-deployer-test-helper:0.1.0")
-	assert.NoError(t, err)
+		"quay.io/arcalot/podman-deployer-test-helper:0.1.0"))
+	t.Cleanup(func() { assert.NoError(t, container.Close()) })
+
+	// Tell the plugin to output the contents of the mapped file.
+	assert.NoErrorR[int](t)(container.Write([]byte("volume\n")))
 
-	var containerInput = []byte("volume\n")
-	_, err = container.Write(containerInput)
-	assert.NoError(t, err)
 	// Note: If it ends up with length zero buffer, restarting the VM may help:
 	// https://stackoverflow.com/questions/71977532/podman-mount-host-volume-return-error-statfs-no-such-file-or-directory-in-ma
```

Comment:
The comment could probably use some attention now that "ends up with length zero buffer", which is, in this code, indistinguishable from "doesn't contain the marker text" is an expected condition in the `!expectedPass` case ...

Reply:
I made some insubstantial changes to this comment text, but I added a reference to it in the subsequent comment in the code.

Note that in the "successful failure case" we _do_ get a non-zero-length buffer back -- it contains the command prompt and some subsequent whitespace -- so the zero-length failure _would_ be distinguishable as a "test harness failure" from the otherwise "expected failure".

However, I'm not inclined to chase that:  really, we should have the plugin reporting much more state (like success/failure of the requested operation, as well as success of the overall plugin's execution).  But, that feels out of scope for the present PR.

So, I'm leaving this comment largely as it is, as a note to a future developer who comes to debug a failure case resulting from a zero-length buffer (if, in fact, this particular mode of failure even still exists).